### PR TITLE
HPCC-26689 SuperFile Details

### PR DESCRIPTION
**Pull request by jeclrsg**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/FileDetails.tsx [hunk 1]
```diff
@@ -76,7 +90,7 @@ export const FileDetails: React.FunctionComponent<FileDetailsProps> = ({
             <PivotItem headerText={nlsHPCC.Graphs} itemKey="Graphs" itemCount={file?.Graphs?.ECLGraph?.length} headerButtonProps={{ disabled: isDFUWorkunit }} style={pivotItemStyle(size, 0)}>
                 <FileDetailsGraph cluster={cluster} logicalFile={logicalFile} />
             </PivotItem>
-            <PivotItem headerText={nlsHPCC.Workunit} itemKey="Workunit" style={pivotItemStyle(size, 0)}>
+            <PivotItem headerText={nlsHPCC.Workunit} itemKey="Workunit" style={pivotItemStyle(size, 0)} headerButtonProps={!!file?.isSuperfile ? { disabled: true, style: { color: theme.palette.neutralTertiary } } : {}}>
```
Comment (jeclrsg) [inline esp/src/src-react/components/FileDetails.tsx:93]: @GordonSmith This isn't very noticeable perhaps, but I'm not sure why this color (theme.palette.neutralTertiary) is off. It seems to be defined as #d1d1d1 in src/src-react/themes.ts, which is the same color that is used for Buttons when disabled. But used here, the PivotItem's button is getting a text color of #a19f9d. Have you noticed any problems with colors like this elsewhere?

File: esp/src/src-react/components/FileDetails.tsx [hunk 2]
```diff
@@ -44,10 +46,17 @@ export const FileDetails: React.FunctionComponent<FileDetailsProps> = ({
 
     const isDFUWorkunit = file?.Wuid?.length && file?.Wuid[0] === "D";
 
+    const theme = getTheme();
+
     return <SizeMe monitorHeight>{({ size }) =>
         <Pivot overflowBehavior="menu" style={{ height: "100%" }} selectedKey={tab} onLinkClick={evt => pushUrl(`/files/${cluster}/${logicalFile}/${evt.props.itemKey}`)}>
             <PivotItem headerText={nlsHPCC.Summary} itemKey="summary" style={pivotItemStyle(size)}>
-                <FileSummary cluster={cluster} logicalFile={logicalFile} />
+                {file && file?.isSuperfile !== undefined
```
Comment (GordonSmith) [inline esp/src/src-react/components/FileDetails.tsx:54]: No need for `file &&`

File: esp/src/src-react/components/FileDetails.tsx (hunk 1 above)
Comment (GordonSmith) [reply esp/src/src-react/components/FileDetails.tsx:93]: I have noticed that disabled tabs aren't very pronounced.  I will take a look

File: esp/src/src-react/components/SuperFileSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        setProtected(file?.ProtectList?.DFUFileProtect?.length > 0 || false);
+        setRestricted(file?.IsRestricted || false);
+    }, [file]);
+
+    const [DeleteConfirm, setShowDeleteConfirm] = useConfirm({
+        title: nlsHPCC.Delete,
+        message: nlsHPCC.DeleteSuperfile,
+        onSubmit: React.useCallback(() => {
+            const subfiles = (file?.subfiles?.Item || []).map(s => { return { Name: s }; });
+            WsDfu.SuperfileAction("remove", file.Name, subfiles, true).then(() => replaceUrl("/files"));
```
Comment (GordonSmith) [inline esp/src/src-react/components/SuperFileSummary.tsx:40]: Should have a catch?

File: esp/src/src-react/components/SuperFileSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        { key: "divider_1", itemType: ContextualMenuItemType.Divider, onRender: () => <ShortVerticalDivider /> },
+        {
+            key: "save", text: nlsHPCC.Save, iconProps: { iconName: "Save" }, disabled: !canSave,
+            onClick: () => {
+                file?.update({
+                    UpdateDescription: true,
+                    FileDesc: description,
+                    Protect: _protected ? "1" : "2",
+                    Restrict: restricted ? "1" : "2",
+                });
```
Comment (GordonSmith) [inline esp/src/src-react/components/SuperFileSummary.tsx:68]: Should have a catch

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**